02_knight_and_princess/dana_stepina/main.cpp

```diff
@@ -10,38 +11,37 @@ using std::endl;
 using VertexId = int;
 using EdgeId = int;
 
-constexpr int INVALID_ID = -1;
-const bool TRUE_MEANING = true;
-const bool FALSE_MEANING = false;
-
 struct Vertex {
-  const VertexId id = INVALID_ID;
+  const VertexId id = 0;
+
+  explicit Vertex(const VertexId& vertex_id) : id(vertex_id) {}
 
-  const bool& has_edge(const EdgeId& edge_id) {
-    for (const auto id : edge_ids_)
-      if (edge_id == id)
-        return TRUE_MEANING;
-      else
-        continue;
-    return FALSE_MEANING;
-  }
-  const std::vector<EdgeId>& get_edge_ids() const { return edge_ids_; }
   void add_edge_id(const EdgeId& edge_id) {
-    assert(has_edge(edge_id) && "Such an edge already exists!");
+    assert(!has_edge_id(edge_id) && "Such an edge already exists!");
     edge_ids_.push_back(edge_id);
   }
-  Vertex(VertexId id) : id(id) {}
+  const std::vector<EdgeId>& get_edge_ids() const { return edge_ids_; }
+  bool has_edge_id(const EdgeId& edge_id) {

```

Comment:
Методы, которые не изменяют родительский объект, должны быть помечены как `const`
```suggestion
  bool has_edge_id(const EdgeId& edge_id) const {
```

Reply:
Не вижу, что исправила

Reply:
Все ещё не исправлено...